sdk/go/common/resource/plugin/host.go

```diff
@@ -189,9 +191,18 @@ func parsePluginOpts(providerOpts workspace.PluginOptions, k workspace.PluginKin
 		return handleErr("provider folder '%s' is not a directory", providerOpts.Path)
 	}
 
+	// The path is relative to the project root. Make it absolute here so we don't need to track that everywhere its used.
+	path := providerOpts.Path
+	if !filepath.IsAbs(path) {
+		path, err = filepath.Abs(filepath.Join(root, path))
+		if err != nil {
+			return handleErr("getting absolute path for plugin path %s: %w", providerOpts.Path, err)
+		}
+	}
+
```

Comment:
This is the main fix here. Make sure we resolve any relative plugin paths against the root directory containing the Pulumi.yaml file.